rclpy/rclpy/node.py

```diff
@@ -217,12 +229,85 @@ def get_logger(self):
         """Get the nodes logger."""
         return self._logger
 
+    def declare_parameter(self, parameter: Parameter) -> Parameter:
+        """
+        Declare and initialize a parameter.
+
+        This method, if successful, will result in any callback registered with
+        :func:`set_parameters_callback` to be called.
+
+        :param parameter: The parameter to declare, with a name, value and descriptor.
+        :return: Parameter with the effectively assigned value.
+        :raises: ParameterAlreadyDeclaredException if the parameter had already been declared.
+        :raises: InvalidParameterException if the parameter name is invalid.
+        :raises: InvalidParameterValueException if the registered callback rejects the parameter.
+        """
+        return self.declare_parameters('/', [parameter])
+
+    def declare_parameters(self, namespace: str, parameters: List[Parameter]) -> List[Parameter]:
+        """
+        Declare a list of parameters.
+
+        This method, if successful, will result in any callback registered with
+        :func:`set_parameters_callback` to be called once for each parameter.
+        If one of those calls fail, an exception will be raised and the remaining parameters will
+        not be declared. Parameters declared up to that point will not be undeclared.
+
+        :param namespace: Namespace for parameters.
+        :param parameters: The parameters to declare, with a name, value and descriptor.
+        :return: Parameter list with the effectively assigned values for each of them.
+        :raises: ParameterAlreadyDeclaredException if the parameter had already been declared.
+        :raises: InvalidParameterException if the parameter name is invalid.
+        :raises: InvalidParameterValueException if the registered callback rejects any parameter.
+        """
+        for parameter in parameters:
+            parameter.name = namespace + parameter.name
+            # Note(jubeira): declare_parameters verifies the name, but set_parameters doesn't.
+            validate_parameter_name(parameter.name)
+
+        parameters_already_declared = (
+            parameter.name in self._parameters for parameter in parameters
+        )
+        if any(parameters_already_declared):
+            raise ParameterAlreadyDeclaredException(list(parameters_already_declared))
+
+        # Call the callback once for each of the parameters, using method that doesn't
+        # check whether the parameter was declared beforehand or not.
+        self._set_parameters(parameters, raise_on_failure=True)
+        return self.get_parameters([parameter.name for parameter in parameters])
+
+    def undeclare_parameter(self, name: str):
+        """
+        Undeclare a previously declared parameter.
+
+        This method will not cause a callback registered with
+        :func:`set_parameters_callback` to be called.
+
+        :param name: name of the parameter.
```

Comment:
I realize it wasn't explicit in `rclcpp`, but it might be good to clarify here that the name is fully-qualified (including any namespace provided when declared).

Reply:
Addressed in https://github.com/ros2/rclpy/pull/325/commits/0925fe4e4a72a4b32ff1ec58790d73c4ba35838a.